core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -3248,6 +3253,21 @@ private void setAclInternal(RpcContext rpcContext, SetAclAction action, LockedIn
     targetInode.setLastModificationTimeMs(opTimeMs);
   }
 
+  private void setAclInternal(RpcContext rpcContext, SetAclAction action, LockedInodePath inodePath,
+      List<AclEntry> entries, boolean replay, long opTimeMs, SetAclOptions options)
+      throws IOException, FileDoesNotExistException {
+    try (LockedInodePathList children = options.getRecursive()
+        ? mInodeTree.lockDescendants(inodePath, InodeTree.LockMode.WRITE) : null) {
+      if (children != null) {
+        for (LockedInodePath child : children.getInodePathList()) {
+          setAclSingleInode(rpcContext, action, child, entries, replay, opTimeMs);
+        }
+      }
+    }
+    setAclSingleInode(rpcContext, action, inodePath, entries, replay, opTimeMs);
```

Comment:
should we set the inode one first, and then do the children?

Reply:
done